presto-spark-base/src/main/java/com/facebook/presto/spark/PrestoSparkQueryExecutionFactory.java

```diff
@@ -386,13 +383,14 @@ private PrestoSparkQueryExecution(
                         tableWriteInfo);
                 SerializedPrestoSparkTaskDescriptor serializedTaskDescriptor = new SerializedPrestoSparkTaskDescriptor(sparkTaskDescriptorJsonCodec.toJsonBytes(taskDescriptor));
 
-                JavaPairRDD<Integer, PrestoSparkRow> rdd = createRdd(getOnlyElement(root.getChildren()));
+                SubPlan child = getOnlyElement(root.getChildren());
```

Comment:
Would it be possible that the root has multiple children in some cases? -- e.g. after push table write through union:

```
         TableCommit
                  |
           RemoteSource
            /          \
      Fragment1.    Fragment2
      TableWrite    TableWrite
```

We don't need to fix this in this PR. Just keep in mind :)

Reply:
I remove this limitation when implementing union.